### PR TITLE
doc: can_api: Update the CAN documentation after API update

### LINKED ISSUE #31379: Update CAN-API Documentation
**Describe the bug**
Current [CAN-API](https://docs.zephyrproject.org/latest/reference/networking/can_api.html) mention  wrong structure parameters in `zcan_filter`, (for example `std_id`):

```
const struct zcan_filter my_filter = {
        .id_type = CAN_STANDARD_IDENTIFIER,
        .rtr = CAN_DATAFRAME,
        .std_id = 0x120,
        .rtr_mask = 1,
        .std_id_mask = 0x7F0
};
```

**To Reproduce**
Compare current [CAN-API](https://docs.zephyrproject.org/latest/reference/networking/can_api.html) with [doxygen docs](https://docs.zephyrproject.org/apidoc/latest/group__can__interface.html#structzcan__filter)

**Expected behavior**
Right names in structures

**Additional context**
Placeholder for me for later fix or somebody, willing update it.

### PR DESCRIPTION
**Pull request by alexanderwachter**
This commit is the documentation update after the API rework.

### CODE REVIEW

File: doc/reference/networking/can_api.rst [hunk 1]
```diff
@@ -145,10 +154,10 @@ a mailbox. When a transmitting mailbox is assigned, sending cannot be canceled.
...
           .rtr = CAN_DATAFRAME,
-          .ext_id = 0x123,
+          .id = 0x123,
           .dlc = 8
   };
-  struct device *can_dev;
+  const struct device *can_dev;
   int ret;
 
   frame.data = {1,2,3,4,5,6,7,8};
```
Comment (Nukersson) [inline doc/reference/networking/can_api.rst:163]:
```suggestion
  frame.data[0] = 1;
  frame.data[1] = 2;
  frame.data[2] = 3;
  frame.data[3] = 4;
  frame.data[4] = 5;
  frame.data[5] = 6;
  frame.data[6] = 7;
  frame.data[7] = 8;
```

Not sure, if it is my issue, but with this line I get: 

```
error: expected expression before '{' token
  <line_number> |  frame.data = {1, 2, 3, 4, 5, 6, 7, 8};
```

My suggestion fixes it.

File: doc/reference/networking/can_api.rst [hunk 2]
```diff
@@ -280,14 +289,14 @@ The filter for this example is configured to match the extended identifier
...
+          .id = 0x1234567,
           .rtr_mask = 1,
-          .std_id_mask = CAN_EXT_ID_MASK
+          .id_mask = CAN_EXT_ID_MASK
   };
   CAN_DEFINE_MSGQ(my_can_msgq, 2);
   struct zcan_frame rx_frame;
   int filter_id;
-  struct device *can_dev;
+  const struct device *can_dev;
```
Comment (Nukersson) [inline doc/reference/networking/can_api.rst:299]: Does `const` needed here everywhere?

**Review by KozhinovAlexander — CHANGES_REQUESTED** [review]
Thank you for very fast fix.

File: doc/reference/networking/can_api.rst (hunk 1 above)
Comment (alexanderwachter) [reply doc/reference/networking/can_api.rst]: Moved it to initializer.

**Review by KozhinovAlexander — APPROVED** [review]
Sorry. Added more changes

File: doc/reference/networking/can_api.rst [hunk 3]
```diff
@@ -145,14 +154,13 @@ a mailbox. When a transmitting mailbox is assigned, sending cannot be canceled.
...
+          .dlc = 8,
+          .data = {1,2,3,4,5,6,7,8}
   };
-  struct device *can_dev;
+  const struct device *can_dev;
   int ret;
 
-  frame.data = {1,2,3,4,5,6,7,8};
-
   can_dev = device_get_binding("CAN_0");
```
Comment (Nukersson) [inline doc/reference/networking/can_api.rst:164]:
```suggestion
  can_dev = device_get_binding(DT_CHOSEN_ZEPHYR_CAN_PRIMARY_LABEL);
```

**Review by KozhinovAlexander — CHANGES_REQUESTED** [review]

File: doc/reference/networking/can_api.rst [hunk 4]
```diff
@@ -221,12 +231,12 @@ The filter for this example is configured to match the identifier 0x123 exactly.
...
+          .id = 0x123,
           .rtr_mask = 1,
-          .std_id_mask = CAN_STD_ID_MASK
+          .id_mask = CAN_STD_ID_MASK
   };
   int filter_id;
-  struct device *can_dev;
+  const struct device *can_dev;
 
   can_dev = device_get_binding("CAN_0");
```
Comment (Nukersson) [inline doc/reference/networking/can_api.rst:241]:
```suggestion
  can_dev = device_get_binding(DT_CHOSEN_ZEPHYR_CAN_PRIMARY_LABEL);
```

File: doc/reference/networking/can_api.rst (hunk 2 above)
Comment (alexanderwachter) [reply doc/reference/networking/can_api.rst:300]: Jep. `const struct device * device_get_binding(const char * name)` returns a const device pointer.

File: doc/reference/networking/can_api.rst [hunk 5]
```diff
@@ -252,13 +262,13 @@ The filter for this example is configured to match a filter range from
...
           .rtr_mask = 1,
-          .std_id_mask = 0x7F0
+          .id_mask = 0x7F0
   };
   struct zcan_work rx_work;
   int filter_id;
-  struct device *can_dev;
+  const struct device *can_dev;
 
   can_dev = device_get_binding("CAN_0");
```
Comment (Nukersson) [inline doc/reference/networking/can_api.rst:273]:
```suggestion
  can_dev = device_get_binding(DT_CHOSEN_ZEPHYR_CAN_PRIMARY_LABEL);
```

File: doc/reference/networking/can_api.rst [hunk 6]
```diff
@@ -280,14 +290,14 @@ The filter for this example is configured to match the extended identifier
...
-          .std_id_mask = CAN_EXT_ID_MASK
+          .id_mask = CAN_EXT_ID_MASK
   };
   CAN_DEFINE_MSGQ(my_can_msgq, 2);
   struct zcan_frame rx_frame;
   int filter_id;
-  struct device *can_dev;
+  const struct device *can_dev;
 
   can_dev = device_get_binding("CAN_0");
```
Comment (Nukersson) [inline doc/reference/networking/can_api.rst:302]:
```suggestion
  can_dev = device_get_binding(DT_CHOSEN_ZEPHYR_CAN_PRIMARY_LABEL);
```

File: doc/reference/networking/can_api.rst [hunk 7]
```diff
@@ -308,6 +318,41 @@ The filter for this example is configured to match the extended identifier
...
+a bitrate and sampling point in permille. The following example sets the bitrate
+to 250k baud with the sampling point at 87.5%.
+
+.. code-block:: C
+
+  struct can_timing timing;
+  const struct device *can_dev;
+  int ret;
+
+  can_dev = device_get_binding("CAN_0");
```
Comment (Nukersson) [inline doc/reference/networking/can_api.rst:339]:
```suggestion
  can_dev = device_get_binding(DT_CHOSEN_ZEPHYR_CAN_PRIMARY_LABEL);
```

**Review by KozhinovAlexander — CHANGES_REQUESTED** [review]
Sorry. Some more changes.

File: doc/reference/networking/can_api.rst (hunk 3 above)
Comment (alexanderwachter) [reply doc/reference/networking/can_api.rst:164]: The code snipped should be as simple as possible. No macro magic.

**Review by KozhinovAlexander — APPROVED** [review]
LGTM and many thanks. Do you have experience with FD-CAN?

**Comment by alexanderwachter** [comment]
@Nukersson CAN-FD drivers are WIP.

**Comment by Nukersson** [comment]
> @Nukersson CAN-FD drivers are WIP.

Are you working on CAN-FD stack or writing also drivers?

**Comment by alexanderwachter** [comment]
@Nukersson lets move this discussion to Slack or the the mailing-list.

Slack invite link:
https://tinyurl.com/y5glwylp

**Review by nashif — APPROVED** [review]

**Pull request merged**